pyflakes/checker.py

```diff
@@ -1129,6 +1130,10 @@ def addBinding(self, node, value):
                 break
         existing = scope.get(value.name)
 
+        in_global_scope = value.name in self.scopeStack[-1]
+        if existing and in_global_scope and existing.isUnassigned:
+            existing.isUnassigned = False
```

Comment:
I don't think this is correct -- for instance a closure-scoped variable can shadow a global variable (and is a separate scope)

Reply:
sweet, added a test case for that